core/db_classes/EE_Payment.class.php

```diff
@@ -700,12 +700,23 @@ protected function _args_as_inputs($args)
         $html = '';
         if ($args !== null && is_array($args)) {
             foreach ($args as $name => $value) {
-                $html .= EEH_HTML::nl(0)
+                if (is_array($value)) {
+                    foreach ($value as $array_value) {
+                        $html .= EEH_HTML::nl(0)
+                             . '<input type="hidden" name="'
+                             . $name
+                             . '[]" value="'
+                             . esc_attr($array_value)
+                             . '"/>';
+                    }
+                } else {
+                    $html .= EEH_HTML::nl(0)
                          . '<input type="hidden" name="'
                          . $name
                          . '" value="'
                          . esc_attr($value)
                          . '"/>';
+                }
```

Comment:
how about extract this logic into a new method that can then recurse over values if passed an array:

```suggestion
                
				$html .= $this->generateInput($name, $value);
```

then add this below the current method:

```php
			private function generateInput($name, $value)
    		{
        		if (is_array($value)) {
            		$html = '';
            		$name = "{$name}[]";
            		foreach ($value as $array_value) {
                		$html .= $this->generateInput($name, $array_value);
            		}
            		return $html;
       			}
        		return EEH_HTML::nl()
               		. '<input type="hidden" name="' . $name . '"'
               		. ' value="' . esc_attr($value) . '"/>';
    		}
```

which has less duplication and means any future changes needed for that inputcan be made in one place